pkg/perf/monitor.go

```diff
@@ -0,0 +1,100 @@
+package perf
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/go-redis/redis"
+	"github.com/jasonlvhit/gocron"
+	"github.com/pkg/errors"
+)
+
+// TaskMethod is the task method signature
+type TaskMethod func(ctx context.Context) (interface{}, error)
+
+type Task struct {
+	Key            string
+	Method         TaskMethod
+	Interval       time.Duration // interval in seconds
+	ExecutionCount uint64
+}
+
+// PerformanceMonitor holds the module data
+type PerformanceMonitor struct {
+	Scheduler   *gocron.Scheduler
+	RedisClient *redis.Client
+	Tasks       []Task
+}
+
+// NewPerformanceMonitor returns PerformanceMonitor instance
+func NewPerformanceMonitor(redisAddr string) *PerformanceMonitor {
+	redisClient := redis.NewClient(&redis.Options{
+		Network: "unix",
+		Addr:    redisAddr,
+	})
+
+	scheduler := gocron.NewScheduler()
+
+	return &PerformanceMonitor{
+		Scheduler:   scheduler,
+		RedisClient: redisClient,
+		Tasks:       []Task{},
+	}
+}
+
+// AddTask a simple helper method to add new tasks
+func (pm *PerformanceMonitor) AddTask(taskName string, interval time.Duration, task TaskMethod) {
+	pm.Tasks = append(pm.Tasks, Task{
+		Key:            taskName,
+		Method:         task,
+		Interval:       interval,
+		ExecutionCount: 0,
+	})
+}
+
+// InitScheduler adds all the test to the scheduler queue
+func (pm *PerformanceMonitor) InitScheduler() {
+	pm.AddTask("TestLogging", 5, TestLogging)
+}
+
+// RunScheduler adds the tasks to the corn queue and start the scheduler
+func (pm *PerformanceMonitor) RunScheduler(ctx context.Context) error {
```

Comment:
the name RunScheduler shouldn't be introduced, a simple Run or Start should have been enough, now whoever is using this API will need to know it's using go-cron scheduler